src/reactive/consumer.py

```diff
@@ -56,3 +60,26 @@ def consume(mongodb_uri: str, queue_name: str) -> None:
                 job_details.run_url,
             )
             msg.ack()
+
+
+def _add_sigterm_handler(queue: SimpleQueue, signal_code: signal.Signals) -> None:
+    """Add a signal handler to clean up and exit.
+
+    Args:
+        queue: The queue to clean up.
+        signal_code: The signal code to handle.
+    """
+
+    def sigterm_handler(signal_code: int, _: FrameType | None) -> None:
+        """Handle a signal.
+
+        Requeue the message and exit.
+
+        Args:
+            signal_code: The signal code to handle.
+        """
+        logger.error("Signal '%s' received. Requeueing message.", signal.strsignal(signal_code))
+        queue.channel.basic_recover(requeue=True)
```

Comment:
Would this have some sort of a timeout?

Reply:
Unfortunately there is no option to pass a timeout, but digging into the lib it seems that pymongo is being used, which seems to use some sort of timeout by default (just skimmed over it, so its more of a guess).